Nos/Views/Components/Images/ImageViewer.swift

```diff
@@ -0,0 +1,230 @@
+import SDWebImageSwiftUI
+import SwiftUI
+
+/// A viewer for images. Supports full-screen zoom and panning.
+struct ImageViewer: View {
+    /// The URL of the image to display.
+    let url: URL
+
+    @Environment(\.dismiss) private var dismiss
+    
+    /// The current zoom scale of the image.
+    @State private var scale: CGFloat = 1.0
+
+    /// The offset of the image. This is updated when the user has zoomed and is panning up, down, left, and right.
+    @State private var offset: CGSize = .zero
+
+    /// The previous offset of the image.
+    /// - SeeAlso: `offset`
+    @State private var zoomScale: CGFloat = 1.0
+
+    /// The size of the image. Will be set to a non-zero value when the image has loaded.
+    @State private var imageSize: CGSize = .zero
+    
+    /// The maximum zoom scale for the image.
+    private let maxZoomScale: CGFloat = 10.0
+    
+    /// The minimum zoom scale for the image.
+    private let minZoomScale: CGFloat = 1.0
+
+    var body: some View {
+        ZStack {
+            Color.imageBackground
+
+            ZoomableContainer {
+                WebImage(url: url)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+            }
+
+            ZStack(alignment: .topLeading) {
+                Color.clear
+
+                Button {
+                    dismiss()
+                } label: {
+                    Image(systemName: "xmark")
+                        .symbolVariant(.fill.circle)
+                        .symbolRenderingMode(.palette)
+                        .foregroundStyle(Color.buttonCloseForeground, Color.buttonCloseBackground)
+                        .font(.title)
+                }
+                .padding()
+            }
+        }
+        .ignoresSafeArea()
+    }
+
+    func resetImageState() {
+        withAnimation(.interactiveSpring()) {
+            zoomScale = minZoomScale
+        }
+    }
+
+    func onImageDoubleTapped(value: TapGesture.Value) {
+        if zoomScale == minZoomScale {
+            withAnimation(.spring()) {
+                zoomScale = 4
```

Comment:
I would like to see this number (4) declared in a constant

Reply:
I think you have a maxZoomScale

Reply:
This isn't even used -- it was from another implementation. Removed!